src/components/landing/layout/footer/language-picker.tsx

```diff
@@ -1,62 +1,69 @@
 import moment from 'moment'
-import React from 'react'
+import React, { useCallback } from 'react'
 import { Form } from 'react-bootstrap'
 import { useTranslation } from 'react-i18next'
 
+const languages = {
+  en: 'English',
+  'zh-CN': '简体中文',
+  'zh-TW': '繁體中文',
+  fr: 'Français',
+  de: 'Deutsch',
+  ja: '日本語',
+  es: 'Español',
+  ca: 'Català',
+  el: 'Ελληνικά',
+  pt: 'Português',
+  it: 'Italiano',
+  tr: 'Türkçe',
+  ru: 'Русский',
+  nl: 'Nederlands',
+  hr: 'Hrvatski',
+  pl: 'Polski',
+  uk: 'Українська',
+  hi: 'हिन्दी',
+  sv: 'Svenska',
+  eo: 'Esperanto',
+  da: 'Dansk',
+  ko: '한국어',
+  id: 'Bahasa Indonesia',
+  sr: 'Cрпски',
+  vi: 'Tiếng Việt',
+  ar: 'العربية',
+  cs: 'Česky',
+  sk: 'Slovensky'
+}
+
+const findLanguageCode = (wantedLanguage: string): string => {
+  let foundLanguage = Object.keys(languages).find((supportedLanguage) => wantedLanguage === supportedLanguage)
+  if (!foundLanguage) {
+    foundLanguage = Object.keys(languages).find((supportedLanguage) => wantedLanguage.substr(0, 2) === supportedLanguage)
+  }
+  return foundLanguage || ''
```

Comment:
How about:

```
return languages[wantedLanguage] || languages[wantedLanguage.substr(0, 2)] || ''
```

That would save two O(f(n))¹ functions and replace it with an O(1) function.

¹ f(n) is the search functions complexity for n.

Reply:
Well, valid point. Feel free to open a PR :yum: